tests/plugins/test_push_floating_tags.py

```diff
@@ -0,0 +1,477 @@
+    }
+
+    temp_dir = mkdtemp(dir=str(tmpdir))
+    with open(os.path.join(temp_dir, ".dockercfg"), "w+") as dockerconfig:
+        dockerconfig_contents = {
+            REGISTRY_V2: {
+                "username": "user", "password": DOCKER0_REGISTRY
+            }
+        }
+        dockerconfig.write(json.dumps(dockerconfig_contents))
+        dockerconfig.flush()
+        all_registry_conf[REGISTRY_V2]['secret'] = temp_dir
+
+    registry_conf = {
+        k: v for k, v in all_registry_conf.items() if k in registries
+    }
+
+    plugins_conf = [{
+        'name': PushFloatingTagsPlugin.key,
+        'args': {
+            'registries': registry_conf,
+        },
+    }]
+
+    mocked_registries, annotations = mock_registries(registry_conf, workers,
+                                                     primary_images=primary_images,
+                                                     manifest_results=manifest_results,
+                                                     schema_version=schema_version)
+    tasker, workflow = mock_environment(tmpdir, primary_images=primary_images,
+                                        floating_images=floating_tags,
+                                        manifest_results=manifest_results,
+                                        annotations=annotations)
+
+    if workers:
+        workflow.buildstep_plugins_conf = [{'name': PLUGIN_BUILD_ORCHESTRATE_KEY}]
+
+    if reactor_config_map:
+        registries_list = []
+
+        for docker_uri in registry_conf:
+            reg_ver = registry_conf[docker_uri]['version']
+            reg_secret = None
+            if 'secret' in registry_conf[docker_uri]:
+                reg_secret = registry_conf[docker_uri]['secret']
+
+            new_reg = {}
+            if reg_secret:
+                new_reg['auth'] = {'cfg_path': reg_secret}
+            else:
+                new_reg['auth'] = {'cfg_path': str(temp_dir)}
+            new_reg['url'] = 'https://' + docker_uri + '/' + reg_ver
+
+            registries_list.append(new_reg)
+
+        platform_descriptors_list = []
+        for platform in goarch:
+            new_plat = {
+                'platform': platform,
+                'architecture': goarch[platform],
+            }
+            platform_descriptors_list.append(new_plat)
+
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+        workflow.plugin_workspace[ReactorConfigPlugin.key][WORKSPACE_CONF_KEY] =\
+            ReactorConfig({'version': 1,
+                           'registries': registries_list,
+                           'platform_descriptors': platform_descriptors_list})
+
+    runner = ExitPluginsRunner(tasker, workflow, plugins_conf)
+    results = runner.run()
+    plugin_result = results[PushFloatingTagsPlugin.key]
+
+    if expected_exception is None:
+        _, list_type = {
+            'v2': (
+                'application/vnd.docker.distribution.manifest.v2+json',
```

Comment:
We're not using this value.

We probably want:
```
list_type = {
  'v2': 'application/vnd.docker.distribution.manifest.list.v2+json',
  'oci': 'application/vnd.oci.image.index.v1+json',
}[schema_version]
```

Reply:
if we remove lines 459-467, this needs to go away entirely.

Fixed.